reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLog.java

```diff
@@ -19,15 +19,22 @@
 import reactor.util.Loggers;
 
 import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Predicate;
 
 /**
  * Log the http access information.
+ * All logs are written to a Logger named {@code reactor.netty.http.server.AccessLog} at INFO level.
+ * <p>
+ * This class also exposes convenience methods to create an access log factory to be passed to {@link reactor.netty.http.server.HttpServer#accessLog(Function)}
+ * during server configuration. Note that access logging must be globally enabled first for that configuration to
```

Comment:
This class has a single purpose to log with the provided logFormat and arguments using a specific logger.
I propose to keep this single purpose and create a new factory class containing the two additional methods.
With this change we will have single and simple purpose for each class.
Wdyt?

Reply:
mmh yeah I see what you mean. `AccessLog` seemed the most logical existing place a user would look for such convenience methods, since the configuration at `HttpServer` level requires a `Function` that produces an `AccessLog`.

We also thought that encapsulating the filtering in the generated `Function` was better than exposing an API that tries too hard to stick to single responsibility principle by directly returning an `AccessLog`, eg.:

```java
//option 1
static public Function<AccessLogArgsProvider, AccessLog> filtering(Predicate<AccessLogArgsProvider> predicate);

//option 2
static public AccessLog filtering(AccessLogArgsProvider args, Predicate<AccessLogArgsProvider> predicate);

HttpServer.create()
    .accessLog(AccessLog.filtering(somePredicate))
    .accessLog(args -> AccessLog.filtering(args, somePredicate))
```

The second option seems redundant, and the goal is to facilitate the writing of the `Function` itself after all...

Also note that while users could arguably write the `somePredicate.test(args)` part and `return null` part themselves, when the predicate matches they'd have no way to mimic the default `AccessLog` pattern:

```java
HttpServer.create()
    .accessLog(args -> somePredicate.test(args) ? new AccessLog(????uh oh????) : null)
```

Reply:
what about to deprecate the current method and to expose 
`.accessLog(*Factory)`, `*Factory` will implement Function and also expose this convenient methods?

Reply:
ok we tried that in the latest commit, but frankly I'm not convinced there is high value in adding an interface (and consequently a configuration method) just for that...

the `AccessLog` class has little to no public responsibility (the only public API is the `create` factory method), so I'd tend to say having it expose factory methods for factory functions is not that much of a stretch (these functions do ultimately represent "a way to create an `AccessLog`", after all).

having a look at the relevant commit, do you still strongly feel there is value in separating these concerns @violetagg?

Reply:
from API point of view it is more obvious to use `accessLog(AccessLogFormat)` in comparison with
`accessLog(Function)` and then checking that actually `AccsessLog` provides also some kind of factory methods that can be used with `accessLog(Function)`

wdyt?

Reply:
yeah, it's marginally better at naturally redirecting users to the correct place for factory methods. ok, let's keep it in `AccessLogFactory` then 👍